[ISSUE]: Azure DevOps Pipeline GitVersion Task with error: "maxBuffer length exceeded"

### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/GitTools/actions/issues) to ensure it has not already been reported

### GitVersion package

AzureDevops task

### What are you seeing?

In Azure yml pipeline step we use the gitversion tasks just like in the usage examples and get following error when executing 

gitversion/execute@3.0.0

This is the output: 

Command: git -C C:/a/15/s rev-parse --is-shallow-repository
Command: C:\hostedtoolcache\windows\GitVersion.Tool\6.0.2\dotnet-gitversion.exe C:/a/15/s /output json /output buildserver /config C:\a\15\s\GitVersion.yml
##[error]stdout maxBuffer length exceeded


### What is expected?

well guess what ? should work 

### Steps to Reproduce

azure devops pipeline with these steps:

steps:
- task: gitversion/setup@3.0.0
  displayName: 'Install GitVersion'
  inputs:
    versionSpec: '6.x'

- task: gitversion/execute@3.0.0
  displayName: 'Configure Git Version'
  name: ${{ parameters.VersionStepName }}
  inputs:
    useConfigFile: true
    configFilePath: ${{ parameters.ConfigFilePath }}

### Output log or link to your CI build (if appropriate).

_No response_

--- a/src/agents/common/build-agent.ts
+++ b/src/agents/common/build-agent.ts
@@ -1,4 +1,4 @@
-import { exec as execNonPromise } from 'node:child_process'
+import { exec as execNonPromise, ExecOptions } from 'node:child_process'
 import * as fs from 'node:fs/promises'
 import * as process from 'node:process'
 import * as path from 'node:path'
@@ -40,6 +40,7 @@ export interface IBuildAgent {
     findLocalTool(toolName: string, versionSpec: string): Promise<string | null>
 
     getInput(input: string, required?: boolean): string
+
     getInput<T>(input: Extract<keyof T, string>, required?: boolean): string
 
     getBooleanInput<T>(input: Extract<keyof T, string>, required?: boolean): boolean
@@ -187,7 +188,7 @@ export abstract class BuildAgentBase implements IBuildAgent {
         const destPath = path.join(cacheRoot, tool, version)
         if (await this.directoryExists(destPath)) {
             this.debug(`Destination directory ${destPath} already exists, removing`)
-            await fs.rm(destPath, { recursive: true, force: true, maxRetries: 3, retryDelay: 1000 })
+            await this.removeDirectory(destPath)
         }
 
         this.debug(`Copying ${sourceDir} to ${destPath}`)
@@ -229,7 +230,8 @@ export abstract class BuildAgentBase implements IBuildAgent {
         const exec = util.promisify(execNonPromise)
 
         try {
-            const { stdout, stderr } = await exec(`${cmd} ${args.join(' ')}`)
+            const commandOptions: ExecOptions = { maxBuffer: 1024 * 1024 * 10 } // 10MB
+            const { stdout, stderr } = await exec(`${cmd} ${args.join(' ')}`, commandOptions)
             return {
                 code: 0,
                 error: null,

